### PR TITLE
OCPBUGS-236: Add missing time units to ingresscontroller.spec.tuningOptions.reloadInterval

### PR DESCRIPTION
**Pull request by gcs278**
The `ingresscontroller.spec.tuningOptions.reloadInterval` is missing time units from the regex pattern which breaks ingress controller deletion  such as when you apply a valid option with a decimal point (`0.5s`) `reloadInterval`. The cluster-ingress-operator uses `time.duration` which converts it to a millisecond value (`500ms`) in it's internal logic. Therefore, cluster-ingress-control is unable to delete ingress controllers with a decimal `reloadInterval` due to this valid conversion.

### CODE REVIEW

File: operator/v1/types_ingress.go [hunk 1]
```diff
@@ -1534,7 +1534,7 @@ type IngressControllerTuningOptions struct {
 	// subsequent reload.
 	//
 	// +kubebuilder:validation:Optional
-	// +kubebuilder:validation:Pattern=^(0|([0-9]+(\.[0-9]+)?(s|m))+)$
+	// +kubebuilder:validation:Pattern=^(0|([0-9]+(\.[0-9]+)?(ms|s|m))+)$
```
Comment (Miciah) [inline operator/v1/types_ingress.go:1537]: https://go.dev/play/p/dgWS_-goMq8 shows that `time.ParseDuration` converts "0.5s10ms10ms" to 520ms, so this seems fine.

**Review by Miciah — COMMENTED** [review]
/approve

File: operator/v1/types_ingress.go (hunk 1 above)
Comment (JoelSpeed) [inline operator/v1/types_ingress.go:1537]: It's probably safer to just support all possible suffixes
```suggestion
	// +kubebuilder:validation:Pattern=^(0|([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h))+)$
```

**Review by JoelSpeed — COMMENTED** [review]
Thanks
/lgtm

**Comment by deads2k** [comment]
> which converts it to a millisecond value (500ms) in it's internal logic.

I'm ok with adding units and loosening the regex, but it sounds like this is the core problem.  If another automated process was controlling this, the two would fight.  

/approve

File: operator/v1/types_ingress.go [hunk 2]
```diff
@@ -1525,16 +1525,17 @@ type IngressControllerTuningOptions struct {
...
+	// Valid time units are "ns", "us" (or "µs" U+00B5 or "μs" U+03BC), "ms", "s", "m", "h".
 	//
 	// Note: Setting a value significantly larger than the default of 5s can cause latency
 	// in observing updates to routes and their endpoints. HAProxy's configuration will
 	// be reloaded less frequently, and newly created routes will not be served until the
 	// subsequent reload.
 	//
 	// +kubebuilder:validation:Optional
-	// +kubebuilder:validation:Pattern=^(0|([0-9]+(\.[0-9]+)?(s|m))+)$
+	// +kubebuilder:validation:Pattern=^(0|([0-9]+(\.[0-9]+)?(ns|us|µs|μs|ms|s|m|h))+)$
```
Comment (Miciah) [inline operator/v1/types_ingress.go:1538]: Why is "µs" included twice? ...Oh, we include it twice in the validation for `spec.tuningOptions.healthCheckInterval` too.

File: operator/v1/types_ingress.go [hunk 3]
```diff
@@ -1525,16 +1525,17 @@ type IngressControllerTuningOptions struct {
 	// A zero value for reloadInterval tells the IngressController to choose the default,
 	// which is currently 5s and subject to change without notice.
 	//
-	// This field expects an unsigned duration string of decimal numbers, each with optional
-	// fraction and a unit suffix, e.g. "100s", "1m30s". Valid time units are "s" and "m".
+	// Expects an unsigned duration string of decimal numbers, each with optional
+	// fraction and a unit suffix, eg "300ms", "1.5h" or "2h45m".
```
Comment (Miciah) [inline operator/v1/types_ingress.go:1529]: Eeeeeurgh...  I kinda prefer "e.g.".  The IBM Style Guide actually recommends spelling out "for example".  I suppose it isn't all that important though.  Why drop "This field"?

File: operator/v1/types_ingress.go (hunk 2 above)
Comment (Miciah) [reply operator/v1/types_ingress.go:1538]: Oh, I should have read the comment above.  I see now:  Those are different code points.  (In my defense, I did a copy and paste and ctrl-f, but it turns out both Firefox and Emacs treat them the same after character folding.)

Comment (gcs278) [reply operator/v1/types_ingress.go:1538]: Yea I just copied and pasted in vim, I suppose that should be good enough.

File: operator/v1/types_ingress.go (hunk 3 above)
Comment (gcs278) [reply operator/v1/types_ingress.go]: I just copied and pasted from another "duration string" field, I'll add it back in this field for consistency and add the periods.

**Review by JoelSpeed — COMMENTED** [review]
/lgtm

**Pull request merged**